docs/learning/ECSQLTutorial/ECSqlSyntax.md

```diff
@@ -0,0 +1,1289 @@
+
+1. `UNION` - take a union of result of two queries such that there is no duplicate results.
+1. `UNION ALL` - take a union of results of two queries.
+1. `INTERSECT` - take only rows that are common in both queries.
+1. `EXCEPT` - take rows from first queries that are not present in second query.
+
+Simple union with no duplicate rows
+
+```sql
+SELECT 1 a ,2 b
+UNION
+SELECT 1 a, 2 b
+/*
+a | b
+------
+1 | 2
+*/
+```
+
+Simple union with duplicate rows
+
+```sql
+SELECT 1 a ,2 b
+UNION ALL
+SELECT 1 a, 2 b
+/*
+a | b
+------
+1 | 2
+1 | 2
+*/
+```
+
+Simple intersect return only common results
+
+```sql
+SELECT 1 a ,2 b
+INTERSECT
+SELECT 1 a, 2 b
+/*
+a | b
+------
+1 | 2
+*/
+```
+
+Except return exclude result from first query by second.
+
+```sql
+SELECT 1 a ,2 b
+EXCEPT
+SELECT 1 a, 2 b
+/*
+a | b
+------
+*/
+```
+
+## Parameters
+
+ECSQL support named and positional parameters.
+
+### Named parameters
+
+Name parameter can be use to bind parameter by name.
+
+Syntax: `:<parameter-name>`
+
+```sql
+    SELECT * FROM meta.ECClassDef WHERE Name = :className
+```
+
+### Positional parameters
+
+Positional parameter are bind by position from left to right.
+
+Syntax: `?`
+
+```sql
+    SELECT * FROM meta.ECClassDef WHERE Name = ? AND DisplayLabel = ?
+```
+
+## ORDER BY clause
+
+Sort result by set of expressions in ascending or descending order. It is also use to order nulls in result set by putting them in front or last of results.
+
+Syntax:
+
+```sql
+ORDER BY
+    <expr> [ASC|DESC] [NULLS FIRST|LAST] [,...]
+```
+
+Order classes by schema and then class name.
+
+```sql
+SELECT * FROM meta.ECClassDef ORDER BY Schema.Id, Name
+```
+
+Order by DisplayLabel but put null values first.
+
+```sql
+SELECT * FROM meta.ECClassDef ORDER BY DisplayLabel NULLS FIRST
+```
+
+## Type filter
+
+Filter `ECClassId` by set of classes in polymorphic or non-polymorphic manner.
+
+Syntax: `<classId> IS [NOT] ( [ALL|ONLY] <class-name>[, ...])`
+
+Select element where it is of type `PUMP` or `PIPE`.
+
+```sql
+    SELECT * FROM Bis.Element WHERE ECClassId IS (plant.PUMP, plant.PIPE)
+```
+
+Select element where it is exactly of type `PUMP` or `PIPE`.
+
+```sql
+    SELECT * FROM Bis.Element WHERE ECClassId IS (ONLY plant.PUMP, ONLY plant.PIPE)
+```
+
+Find all the element that is not of type `PUMP` or `PIPE`
+
+```sql
+    SELECT * FROM Bis.Element WHERE ECClassId IS NOT (plant.PUMP, plant.PIPE)
+```
+
+## Common table expression
+
+Syntax:
+
+```sql
+WITH [RECLUSIVE]
+    <cte-name>([args...]) AS (
+        <query1>
+        [UNION <query2>]
+    )[, <next-cte-block>]
+    <query3>
+```
+
+A simple example of cte.
+
+```sql
+WITH RECURSIVE
+    c(i) AS (
+        SELECT 1
+        UNION
+        SELECT i + 1 FROM c WHERE i < 4 ORDER BY 1
+    )
+    SELECT i FROM c
+    /*
+        i
+        ------------------
+        1
+        2
+        3
+        4
+    */
+```
+
+Generate mandelbrot set using CTE.
```

Comment:
I think it would be better to have a bis example here instead of another synthetic example.  While cool this isn't all that helpful

Reply:
i replaced the example